server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -274,9 +284,20 @@ boolean isRetryableClusterBlockException(final Throwable e) {
 
     private void handleOperationRequest(final Request request, final TransportChannel channel, Task task) {
         Releasable releasable = checkOperationLimits(request);
-        ActionListener<Response> listener =
-            ActionListener.runBefore(new ChannelActionListener<>(channel, actionName, request), releasable::close);
-        runReroutePhase(task, request, listener, false);
+        request.incRef();
```

Comment:
nit: can we move the `incRef` to just before the `try`? I think that makes it easier to avoid something that throws going in between in the future.

Reply:
this was dried away